src/libsystemd/sd-device/device-util.c

```diff
@@ -65,6 +65,31 @@ int device_open_from_devnum(mode_t mode, dev_t devnum, int flags, char **ret) {
         return TAKE_FD(fd);
 }
 
+int device_is_removable(sd_device *device) {
+        const char *subsystem;
+        int r;
+
+        for (;;) {
+                r = device_get_sysattr_bool(device, "removable");
+                if (r >= 0)
+                        return r;
```

Comment:
if this becomes a more generic helper thing we should take greater care of error handling. i.e. any error here that isn't ENOENT should really be propagated as is.